data_structures/src/transaction.rs

```diff
@@ -479,47 +479,50 @@ impl MintTransaction {
         }
     }
 
-    /// Try to create a MintTransaction with number of outputs equal to `utxos_required`,
-    /// where every one of those outputs must be equal to or greater than `collateral_minimum`.
-    /// If the reward is too small, the number of outputs may be smaller than `utxos_required`.
-    /// The output value will only be smaller than `collateral_minimum` if the `reward` is smaller
-    /// than `collateral_minimum`, in which case this function will create a MintTransaction with
-    /// exactly one output
-    pub fn with_split_utxos(
+    /// This method creates a MintTransaction with a reward split in the node and a external address
+    pub fn with_external_address(
         epoch: Epoch,
         reward: u64,
         own_pkh: PublicKeyHash,
-        collateral_minimum: u64,
-        utxos_required: usize,
+        external_address: Option<PublicKeyHash>,
+        external_percentage: u8,
     ) -> Self {
-        let mut vt_outputs = vec![];
         let mut reward = reward;
-        let mut utxo_counter = 1;
-        while reward >= 2 * collateral_minimum && utxo_counter < utxos_required {
-            reward -= collateral_minimum;
-            utxo_counter += 1;
+        let mut vt_outputs = vec![];
+        let mut external_reward = 0;
+        let mut external_pkh = PublicKeyHash::default();
+        if let Some(pkh) = external_address {
+            // In case of a specified pkh, the reward will be redistributed between the node's pkh
+            // and the specified one using the external_percentage value
+            external_reward = reward.saturating_mul(u64::from(external_percentage)) / 100;
+            reward -= external_reward;
+            external_pkh = pkh;
+        }
+        // If `external_percentage` is `100`, the external address will receive the entire
+        // reward, and the output assigning tokens to the node is not needed.
+        if reward > 0 {
             vt_outputs.push(ValueTransferOutput {
                 pkh: own_pkh,
-                value: collateral_minimum,
+                value: reward,
+                time_lock: 0,
+            });
+        }
+        // If `external_percentage` is `0` or there is not any address specified as 'external_address',
+        // the node address will receive the entire reward, and the output assigning tokens to
+        // the external address is not needed.
+        if external_reward > 0 {
+            vt_outputs.push(ValueTransferOutput {
+                pkh: external_pkh,
+                value: external_reward,
                 time_lock: 0,
             })
         }
 
-        vt_outputs.push(ValueTransferOutput {
-            pkh: own_pkh,
-            value: reward,
-            time_lock: 0,
-        });
-
         MintTransaction::new(epoch, vt_outputs)
     }
 
     pub fn len(&self) -> usize {
```

Comment:
This method actually returns the number of transactions, so it it correct to return only 0 or 1. I think that's too misleading, maybe we should remove this method and just inline `if self.is_empty() { 0 } else { 1 }` where it is used.

Reply:
or directly put a 1, because a no mint is an error

Reply:
Probably